### ISSUE TITLE
RFE: flux-module reload command

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#790]
I know it is trivial, but during broker module development it is handy to have a `flux module reload` command to remove/load a module in one command.

I'm currently using a simple reload command like this

``` C
void mod_reload (flux_t h, opt_t opt)
{
    mod_rmmod (h, opt);
    mod_insmod (h, opt);
}

```

which could be submitted as a simplistic PR, but I wonder if it would be better to support a `reload` in the low level Module Extension Protocol [RFC5](https://github.com/flux-framework/rfc/blob/master/spec_5.adoc)? That is definitely _not_ trivial though

**Comment by garlick** [comment]
I'd probably vote for the simple approach for now.

**Comment by garlick** [comment]
A useful feature would be for the "base component" (e.g. broker) to queue messages rather than forward (rank > 0) or respond with ENOSYS (rank == 0) when a reload of an extension module is in progress.  That would argue for a "native" (RFC 5) reload operation.

**Comment by grondo** [comment]
Fixed by #2736
